lib/apiservers/engine/backends/layer_cache.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vicbackends
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/vic/lib/apiservers/portlayer/client/storage"
+	"github.com/vmware/vic/lib/apiservers/portlayer/models"
+	"github.com/vmware/vic/lib/guest"
+)
+
+// LayerCache is an in-memory cache of image layers. It is refreshed at startup
+// by a call to the portlayer. It is updated when new layers are pulled or
+// layers are deleted.
+type LayerCache struct {
+	sync.Mutex
+
+	// cache maps layer ID to layer data
+	cache map[string]*models.Image
```

Comment:
I think I addressed this in the changes I just pushed. Can you take a gander and let me know if this is more in line with what you were thinking?